### ISSUE TITLE
chore: define use-cases to inspire UX design

### ISSUE THREAD

**Issue by sesam** [kalaspuffar/secure-quick-reliable-login#213]
This is a place holder issue to collect a list of use cases, as a first step towards a user-centric design.

*Use cases*

As a user, I want to X in order to Y
1. choose an identity | make a sign-in
2. get started with this app easily | not get side tracked
3. rename identities | not confuse identities
4. track my identity usage | not confuse identities
5. get reminders to keep my identity (ies) properly backed up | not loose access to identities
6. load identity from backup | verify my backup is good
7. see version info | see I'm using the latest software
8. replace identity access password | prevent compromise
9. replace identity | recover from compromise ("rekey" in SQRL jargon, right?)
10. define timeouts & short password usage | trade security for convenience (!)

**Comment by kalaspuffar** [comment]
Sounds like a great idea.

Maybe this diagram of the functions could be of use. Maybe incomplete but I believe everything should be there.

https://github.com/kalaspuffar/secure-quick-reliable-login/blob/master/doc/sqrl.png

**Comment by sesam** [comment]
:+1: I recently looked at https://www.figma.com/file/n5o6ptY5FaPS1E1KmR0Bcx/Grin-web-wallet?node-id=0%3A1 that is a wallet screen flow (a requirement in the ios app dev world). Any designers reading this discussion, per chance?

**Comment by IvarSnaaijer** [comment]
I have made a simple mock-up together with an avid Android user... As Identity is the main point of the app we think it should be on the first page.
The Settings can be divided into Security and identity management.
By allowing some extra room over new screens there can be a bit more explanation on why these features are here but without too much detail.
Also we think that the fontsize must be larger especially on tablets (As big as the numbers are for the rescue code)
[Android SQRL UI.pdf](https://github.com/kalaspuffar/secure-quick-reliable-login/files/2665149/Android.SQRL.UI.pdf)
For this design I also looked at other apps, I was also thinking of using the three bars (left top of most apps) but this became as cluttered as the current ´advanced settings´ screen.

The screens I left out are like the original.
We would like to propose some other colour schemes, the main feature of these will be a normal and dark version. 

Our idea behind the ID selection in Basic security is to allow the user to always default to one specific ID or to automatically remember the last ID.

**Comment by sesam** [comment]
I like the spreading out into a few more screens, which makes the list of possible actions more digestible. For the Manage identities screen, I propose a different order of buttons:

1. Change password
2. Recover
3. Rename
4. Export
5. (visual space)
6. Create new
7. Import

And maybe at some point add an 8. Delete / archive. To be used by testers, and the regular user who likes things clean and just right.

**Comment by kalaspuffar** [comment]
Hi again.

When I looked at the mockup I saw that you had some thought to show identity on the start page. I got the idea that you might present it with images and colors. What you think of this prototype?

https://kalaspuffar.github.io/prototypes/index.html

Use arrow keys to switch identity.

**Comment by kalaspuffar** [comment]
Forgot to say that the background img and color is produced or chosen by the user.

**Comment by IvarSnaaijer** [comment]
I love the simplicity, I can see room for even more simplicity in the future: Make it an option to directly go to the camera when you click the app.
Add the Vertical dot's in the top right in the camera view to get to settings, and If multiple ID's are available show it next to the vertical dots. (selection can probably left out here)

https://drive.google.com/file/d/1Bd_VLs0GU4hSLgJaXN4JQ2UQ0h75bWgr/view?usp=sharing

**Comment by sesam** [comment]
nice!

> go to the camera when you click the app.
> Add the Vertical dot's in the top right in the camera view to get to
> settings
>
I like the idea! Activateable as a checkbox maybe?

   [x] scan when app starts

, and If multiple ID's are available show it next to the vertical dots.
> (selection can probably left out here)
>
Most multi id situations online today use the round profile photo
iconography.

What if several people use one device to login with? This likely what
caused FB and Google to add multi user circle photos online and in Chrome.

We already basically support this with repeating last few chars of the
password for using an already "opened" identity, it would need more
exposure in the UI to be practical to use.

>

**Comment by IvarSnaaijer** [comment]
I added a checkbox in my excuse for a UI design, but Quick Login might not been the best name. 
I suspect I need to make the ´screens´ bigger.

**Comment by kalaspuffar** [comment]
Most of the design I'm ok with. I think I need to see it to get a better feel for how it would work.

Some of the things seem to be a lot of clicks/choices away.

Another thing is the color profile selection. I'm not sure if that is required. I don't like apps without a set theme. Usually, you either like the visual design of an app or you don't.

When it comes to editors like Visual Code that you might use every day for hours and hours then to customize the GUI is more important but an application that has one feature I think multiple profiles is overkill.

I'm open to discussion of what that would be, maybe you don't like purple as much as I do. 

Another thing when it comes to design is to use the standard components in the platform in order to have as few dependencies as possible.

An application is only as secure as the parts in it.

The last consideration is the thought to scan directly and menu to activate the menu. Seems like a really good idea but might require a bit of work. The scanning activity is handled by a third-party library but that might change in the future.

**Comment by IvarSnaaijer** [comment]
My main issue with the current colour screen is the brightness. I set my phone to filter the blue light at night and set a lot of apps to use a white on black scheme. When opening the SQRL app to logon I get my retina's fried ;)

Regarding the 'direct scan' feature. I do not know the real implications for changing this, but it might be a premium feature... You must put a lot of time in this, and when it goes live we need some support. As there are currently no other developers....

**Comment by sesam** [comment]
tl;dr I suggest closing this issue.

@IvarSnaaijer and others, after re-reading the discussion in this ticket, I feel that it's become stale - the current app design has improved enough that I think it covers everything in here.
We (SQRL power users) will likely come up with wordsmithing exercises as new users come in to look at SQRL and at such time I'd suggest we rather start anew.
